pkg/controllermanager/controller/project/stale/reconciler.go

```diff
@@ -151,9 +150,40 @@ func (r *Reconciler) projectInUseDueToBackupEntries(ctx context.Context, namespa
 	return kubernetesutils.ResourcesExist(ctx, r.Client, &gardencorev1beta1.BackupEntryList{}, r.Client.Scheme(), client.InNamespace(namespace))
 }
 
+func (r *Reconciler) projectInUseDueToWorkloadIdentities(ctx context.Context, namespace string) (bool, error) {
+	workloadIdentityList := &metav1.PartialObjectMetadataList{}
+	workloadIdentityList.SetGroupVersionKind(securityv1alpha1.SchemeGroupVersion.WithKind("WorkloadIdentityList"))
+	err := r.Client.List(
+		ctx,
+		workloadIdentityList,
+		client.InNamespace(namespace),
+		gardenerutils.UncontrolledSecretSelector,
+		client.MatchingLabels{v1beta1constants.LabelCredentialsBindingReference: "true"},
+	)
+	if err != nil {
+		return false, err
+	}
+
+	workloadIdentityNames := sets.New[string]()
```

Comment:
I have no hard opinion on this, most of time it should not matter because the number of workloadidentities will be relatively small, so I will leave it up to you whether to change it or not. 
So, the proposal would to allocate the memory required for the set in advance via

```suggestion
	workloadIdentityNames := make(sets.Set[string], len(workloadIdentityList.Items))
```

Reply:
Good suggestion.